src/internal/logger.cpp

```diff
@@ -0,0 +1,81 @@
+#include "logger.hh"
+
+#include <iostream>
+#include <cstdarg>
+#include <chrono>
+#include <iomanip>
+#include <filesystem>
+
+LogLevel Logger::current_level = LogLevel_Info;
+
+void
+Logger::set_log_level(LogLevel level)
+{
+    current_level = level;
+}
+
+LogLevel
+Logger::get_log_level()
+{
+    return current_level;
+}
+
+std::string
+Logger::log(LogLevel level,
```

Comment:
I don't think this code is thread-safe. If multiple threads log messages simultaneously, the output might become interleaved or corrupted.